core/src/main/java/hudson/Util.java

```diff
@@ -1403,7 +1403,7 @@ public static File resolveSymlinkToFile(@Nonnull File link) throws InterruptedEx
      *      The relative path is meant to be resolved from the location of the symlink.
      */
     @CheckForNull
-    public static String resolveSymlink(@Nonnull File link) throws InterruptedException, IOException {
+    public static String resolveSymlink(@Nonnull File link) throws IOException {
```

Comment:
Binary compatible simplification—the body never actually threw that exception.

Reply:
Not source compatible IIRC (javac fails when there is a catch block for impossible exception), may impact PCT. There are usages of it externally, e.g. in the Support Core plugin: https://github.com/search?p=1&q=org%3Ajenkinsci+resolveSymlink&type=Code

I would propose to detach it to a separate PR if you feel it's important enough

Reply:
@oleg-nenashev Jesse already disputed this in https://github.com/jenkinsci/jenkins/pull/3340#issuecomment-372333586

Reply:
The simple workaround for the `buildPlugin` case would be to just `catch (Exception` I guess.